### PR TITLE
add explicit byte encoding to trj files

### PR DESCRIPTION
**Pull request by kain88-de**
fixes python3 errors

PR Checklist
------------
 - [x] Tests?
 ~~- [ ] Docs?~~
~~ - [ ] CHANGELOG updated?~~
 ~~- [ ] Issue raised/referenced?~~

### CODE REVIEW

**Comment by jbarnoud** [comment]
TRJ is an ascii format, you should be able to get the same result just by opening the file in `rt` mode instead on `r`.

**Comment by orbeckst** [comment]
but that's not ascii trj but netcdf

File: package/MDAnalysis/coordinates/TRJ.py [hunk 1]
```diff
@@ -445,8 +445,8 @@ def __init__(self, filename, n_atoms=None, **kwargs):
         self.trjfile = scipy.io.netcdf.netcdf_file(self.filename,
                                                    mmap=self._mmap)
 
-        if not ('AMBER' in self.trjfile.Conventions.split(',') or
-                'AMBER' in self.trjfile.Conventions.split()):
+        if not ('AMBER' in self.trjfile.Conventions.decode('utf-8').split(',') or
+                'AMBER' in self.trjfile.Conventions.decode('utf-8').split()):
             errmsg = ("NCDF trajectory {0} does not conform to AMBER "
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/TRJ.py]: I got an additional related failure in `test_netcdf`:
```
[gw3] darwin -- Python 3.6.1 /Users/oliver/anaconda3/envs/mda/bin/python

self = <testsuite.MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader testMethod=test_metadata>

    def test_metadata(self):
        data = self.universe.trajectory.trjfile
>       assert_equal(data.Conventions, 'AMBER')
E       AssertionError:
E       Items are not equal:
E        ACTUAL: b'AMBER'
E        DESIRED: 'AMBER'
```

The test also needs a `decode('utf-8')`.

**Comment by orbeckst** [comment]
FYI: in `test_netcdf.py`, use
```python
    def test_metadata(self):
        data = self.universe.trajectory.trjfile
        assert_equal(data.Conventions.decode('utf-8'), 'AMBER')
        assert_equal(data.ConventionVersion.decode('utf-8'), '1.0')
```

**Comment by jbarnoud** [comment]
> but that's not ascii trj but netcdf

I looked at the wrong reference file, then.

**Comment by orbeckst** [comment]
>>  but that's not ascii trj but netcdf
> I looked at the wrong reference file, then.

Both Amber ASCII TRJ *and* Amber netCDF code are in the same file, `coordinates/TRJ.py`. This is admittedly a bit confusing and perhaps we ought to split it; it would be more consistent with everything else in `coordinates`.

**Comment by orbeckst** [comment]
I squashed everything in a952501fef185f7eb9f2eaf9f7d89fe6c71d04c6 and updated the comment to reflect that this pertains to Amber netCDF.

**Pull request merged**